### PR TITLE
[CI] Rasa OSS bumper

### PR DESCRIPTION
**Pull request by HotThoughts**
Add a workflow to fetch Rasa Open Source version from rasa repo and bump the version if needed. 

**TODOs**
- [x] Add slack notification on failure
- [x] Add reviewer when creating a PR
- [x] Update README

**Definition of Done**
- [X] The workflow skip steps when this is the newest Rasa OSS version, [example](https://github.com/RasaHQ/rasa-x-demo/pull/41/checks?check_run_id=3171641060#step:5:18) 
- [X] The workflow skip steps when it founds a related PR, [example](https://github.com/RasaHQ/rasa-x-demo/pull/39/checks?check_run_id=3179842823#step:7:20)
- [X] The workflow correctly bump Rasa OSS version and create a PR, [example](https://github.com/RasaHQ/rasa-x-demo/pull/39/checks?check_run_id=3171632284#step:10:11) 
- [X] Label the created PR with `tool:train-and-test`, which will trigger the 'train and test' workflow, see [PR](https://github.com/RasaHQ/rasa-x-demo/pull/41)
- [x] The 'train and test' workflow will train and push the model to the head branch
image

### CODE REVIEW

File: .github/workflows/rasaoss-version-bumper.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        run: |
+          # Fill pull request title and body with branch name and commits  
+          gh pr create --fill \
+            --label "$LABEL" \
+            --reviewer RasaHQ/infrastructure-squad
+
+      - name: Notify slack on failure
+        if: failure()
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
```
Comment (HotThoughts) [inline .github/workflows/rasaoss-version-bumper.yml:101]: We need to add a slack token. I am pretty sure we don't have any in secrets.

File: .github/workflows/rasaoss-version-bumper.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        uses: voxmedia/github-action-slack-notify-build@212e9f7a9ca33368c8dd879d6053972128258985
+        with:
+          channel_id: ${{ secrets.SLACK_ALERTS_CHANNEL_ID }}
```
Comment (HotThoughts) [inline .github/workflows/rasaoss-version-bumper.yml:104]: Also need to add the Channel ID to secrets.

**Comment by HotThoughts** [comment]
This is ready for the first round of reviews. Please 👀 when you have time. @mprazz 
One question, do we want to bump rasa version `2.8.1`?

This workflow created this. I am wondering how to deal with it.
- https://github.com/RasaHQ/rasa-x-demo/pull/41

File: .github/workflows/rasaoss-version-bumper.yml (hunk 2 above)
Comment (tczekajlo) [reply .github/workflows/rasaoss-version-bumper.yml:104]: Done

File: .github/workflows/rasaoss-version-bumper.yml (hunk 1 above)
Comment (tczekajlo) [reply .github/workflows/rasaoss-version-bumper.yml:101]: Done

File: .github/workflows/rasaoss-version-bumper.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          git add -u
+          # Commit & Push
+          git commit -m "Bump Rasa OSS to ${LATEST_RASA_VERSION}"
+          git push origin $BRANCH_NAME
+
+      - name: Open pull request ☄️
+        if: env.CREATE_PR == 'true'
+        env:
+          # This label will trigger train and test workflow
+          LABEL: 'train and test'
```
Comment (tczekajlo) [inline .github/workflows/rasaoss-version-bumper.yml:91]: I think I'd stick to the convention that we have in the other repos and have something like this `tool:train-and-test`, wdyt?

**Review by tczekajlo — COMMENTED** [review]
LGTM, I've left a small comment

**Review by mprazz — APPROVED** [review]
LGTM :medal_military: Let's make sure to share this with the team on the squad sync! :fire:

File: .github/workflows/rasaoss-version-bumper.yml (hunk 3 above)
Comment (HotThoughts) [reply .github/workflows/rasaoss-version-bumper.yml:91]: Good point! Let's make the naming consistent. 👍

File: .github/workflows/rasaoss-version-bumper.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      - name: Open pull request ☄️
+        if: env.CREATE_PR == 'true'
+        env:
+          # This label will trigger train and test workflow
+          LABEL: 'tool:train-and-test' 
+        run: |
+          # Fill pull request title and body with branch name and commits  
+          gh pr create --fill \
+            --label "$LABEL" \
+            --reviewer RasaHQ/infrastructure-squad
```
Comment (HotThoughts) [inline .github/workflows/rasaoss-version-bumper.yml:96]: @mprazz do you know who or which squad should be the *default* reviewer here?

Comment (mprazz) [reply .github/workflows/rasaoss-version-bumper.yml:96]: let's keep it for us right now :sweat_smile:

**Pull request merged**